src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,212 @@
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars_drivers cd "
+                + "JOIN cars c ON c.id = cd.car_id and c.is_deleted = false "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE cd.driver_id = ? ";
+        List<Car> cars = new ArrayList<>();
+        try (Connection driverConnection = ConnectionUtil.getConnection();
+                PreparedStatement statement = driverConnection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars by driver id = "
+                    + driverId, e);
+        }
+        for (Car car: cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    private void insertDriversForCar(Car car) {
+        if (car.getDrivers() == null || car.getDrivers().size() == 0) {
+            return;
+        }
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver: car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't add drivers to car. Car: "
+                    + car + ".", e);
+        }
+    }
+
+    private void deleteDriversForCar(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete drivers from car. Car: "
+                    + car + ".", e);
+        }
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerId, manufacturerName,
+                manufacturerCountry);
```

Comment:
it's better to create a separate method for this logic, so you won't need to add empty lines before and after this block of code

Reply:
ok